### ISSUE TITLE
`create -d . --cfg-proc` is not looking for procedure in reference dataset

### ISSUE THREAD

**Issue by mih** [datalad/datalad#4802]
```
datalad -l debug create -c xnat_dataset -d . sub3
```

`.` is `/tmp/rp`

I expected it to look for `cfg_xnat_dataset` in `/tmp/rp/.datalad/procedures`, but it is not happening. Instead:

````
[DEBUG  ] Looking for procedure 'cfg_xnat_dataset' in '/home/mih/.config/datalad/procedures' 
[DEBUG  ] Looking for procedure 'cfg_xnat_dataset' in '/etc/xdg/datalad/procedures' 
[DEBUG  ] Looking for procedure 'cfg_xnat_dataset' in '/tmp/rp/sub3/.datalad/procedures' 
[DEBUG  ] Looking for procedure 'cfg_xnat_dataset' in '/home/mih/hacking/datalad/neuroimaging/datalad_neuroimaging/resources/procedures' 
[DEBUG  ] Looking for procedure 'cfg_xnat_dataset' in '/home/mih/hacking/datalad/git/datalad/resources/procedures' 
[DEBUG  ] Cannot find procedure with name 'cfg_xnat_dataset' [run_procedure(cfg_xnat_dataset)] 
````

My usecase is a particular configuration that needs to be applied to all subdatasets of a superdataset. It makes sense to me to put such a procedure in the root dataset. I also want to use it with `addurls --cfg-proc`. I have not investigated whether this usage pattern suffers from the same lookup issue.

**Comment by mih** [comment]
Given that #4803 was rejected, here is a summary of the key problem:

I have a dataset that contains a config procedure. I want to make that procedure accessible to `addurls --cfg-proc` for the post-create configuration of subdatasets in a uniform way. In my use case I cannot rely on a specific user config (different users), nor machine config (different machines). All logic and configuration to update a superdataset (and all its subdatasets) should come from the dataset itself, which is only briefly cloned onto a machine for the purpose of updating from a set of generated URLs. Afterwards, the update is pushed and the local clone is destroyed -- and with it all local configuration. This rules out a configuration based on result hooks, as such configuration is not read from the persistent dataset config.

**Comment by yarikoptic** [comment]
@mih please have a look at #5156 and give your blessing or "mneh, since ..."

**Comment by mih** [comment]
The fix in #5156 is not exactly addressing the issue, but it is good enough for now and for me.
